internal/generator/vector/output/splunk/splunk.go

```diff
@@ -63,10 +67,20 @@ func New(id string, o logging.OutputSpec, inputs []string, secret *corev1.Secret
 			Debug(id, vectorhelpers.MakeInputs(inputs...)),
 		}
 	}
+
+	componentID := vectorhelpers.MakeID(id, "add_splunk_index")
 	dedottedID := vectorhelpers.MakeID(id, "dedot")
+
+	dedotInputs := inputs
+	indexRemapElement := SetSplunkIndexRemap(o.Splunk, componentID, inputs)
```

Comment:
If we are to use a remap we should [exclude](https://vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/#encoding.except_fields) the field in the encoding.

Additionally, do we know the behavior if the field is missing? Does the log get dropped?  Does it just use "splunk default" behavior or exclude the index in the submission?

Reply:
The only time a log get's indexed in splunk's `default` index is if the sink omits index.
I use remap so that I can check if an IndexKey is missing from the log, I can then default it to something else as I asked about above. Instead of defaulting to `log_type` we can default it to splunk's `default`

Reply:
The field which I suggested to omit is `write_index` which is the one created by this function.  Per my previous comment for backwards compatibility, I would set nothing